### ISSUE TITLE
[Crash] Only fullscreen activities can request orientation

### ISSUE THREAD

**Issue by Dimezis** [Swrve/swrve-android-sdk#271]
Hi,

There's a crash on Android 8.0.0 with target SDK 27, when trying to set an orientation for `SwrveInAppMessageActivity`. Google introduced this restriction for translucent activities, so they can't request orientation anymore.
I checked your latest sources and it looks like `SwrveInAppMessageActivity` still has this attribute `<item name="android:windowIsTranslucent">true</item>`, and not doing any check before setting the orientation in `onCreate`.

This crash happens when I'm trying to receive some in-app message from ongoing campaign.

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.nadex.dxd.alpha/com.swrve.sdk.messaging.ui.SwrveInAppMessageActivity}: java.lang.IllegalStateException: Only fullscreen activities can request orientation
```

Correct me if I'm missing something. And I can make a PR to take care of this, in case you'd like to.

**Comment by dominicmarmionswrve** [comment]
Great find @Dimezis! We haven't targeted 27 in our latest release but knew there could be implications like this, which is why is kept it at 26 until we thoroughly tested it. If you'd  like to PR it, then that would be great! 🍻  

We're on a code freeze now until after Christmas but please us know if you require a fix for this as soon as possible, or if January is ok? 

Thanks!

**Comment by Dimezis** [comment]
No rush. Anyway, we are still blocked by other issues on SDK 27, and this migration isn't that urgent. 

So I'll make a PR tomorrow maybe. Any rules I need to know about?

**Comment by dominicmarmionswrve** [comment]
Hi @Dimezis, sorry for the delay, but we haven't been able to replicate this issue. Could you give us some steps please?

**Comment by Dimezis** [comment]
I assume that's because you are trying to test it on 8.1.0 phone. I found out that on 8.1.0 they reverted back this restriction after some complains from developers: https://issuetracker.google.com/issues/68454482. 

But this bug still persists on 8.0.0. What you need to do is to create a campaign in Swrve dashboard (In-App Messages), select an orientation there (let's say portrait) and launch the campaign. 

After the campaign being triggered, `SwrveInAppMessageActivity` will be launched and `SwrveMessage message` will be extracted from an intent, with `formats` list in it. This list should contain exactly 1 element. My`SwrveMessageFormat` in this list looks like this: 
![image](https://user-images.githubusercontent.com/1433500/34778422-95572992-f61d-11e7-94d1-bdae9f7f319b.png)

Basically, for testing purposes, you can even avoid launching campaign and just launch this activity  with `SwrveMessage` passed that contains the data I described.
This will crash on Android 8.0.0 with target SDK 27.

**Comment by Dimezis** [comment]
And sorry, didn't have a time to make a PR for it yet

**Comment by dominicmarmionswrve** [comment]
Thanks for the steps! Thats great info! We'll take a look tomorrow and get back to you. 🍻

**Comment by dominicmarmionswrve** [comment]
We've released a hotfix to stop the app crashing on Android 8.0.0 with target SDK 27. We still have testing to do to make sure its fully compatible with api 27 which we'll do with the next minor release.

Thanks for your help on this Dimezis! 🍻

**Comment by hamid290** [comment]
Hi Dominic, we have this issue now on live  here is the log we have the latest sdk version 

Huawei Nexus 6P (angler), Android 8.1
Report 1

java.lang.RuntimeException: 
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2778)
  at android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:2856)
  at android.app.ActivityThread.-wrap11 (Unknown Source)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1589)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loop (Looper.java:164)
  at android.app.ActivityThread.main (ActivityThread.java:6494)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:438)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:807)
Caused by: java.lang.IllegalStateException: 
  at android.os.Parcel.readException (Parcel.java:2012)
  at android.os.Parcel.readException (Parcel.java:1950)
  at android.app.IActivityManager$Stub$Proxy.setRequestedOrientation (IActivityManager.java:5662)
  at android.app.Activity.setRequestedOrientation (Activity.java:5762)
  at com.swrve.sdk.messaging.ui.SwrveInAppMessageActivity.onCreate (SwrveInAppMessageActivity.java:100)
  at android.app.Activity.performCreate (Activity.java:7000)
  at android.app.Activity.performCreate (Activity.java:6991)
  at android.app.Instrumentation.callActivityOnCreate (Instrumentation.java:1214)
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:2731)

**Comment by Sergio-Mira** [comment]
Hi @hamid290 we'll try and reproduce on our side and work on a fix. It is strange tho as that check should be gone from Android 8.1.

Thanks for letting us know!

**Comment by Sergio-Mira** [comment]
I've been testing with the Google 8.1 (API 27) official image in an emulator and InAppMessages display without problems. Do you have any info on how to reproduce it? What kind of OS flavour is that device running?

We could potentially catch the RuntimeException in that section of the code.

**Comment by hamid290** [comment]
Hi Sergio, we are using the release flavuor and the crash so far happen on the testing device with live app, we haven't yet seen any customer having the issue, what we have done to reproduce it's open the app first time, close it and re open it and ti's crash

**Comment by Sergio-Mira** [comment]
Hi @hamid290 we have added code to catch any exception that are raised in that part of the code in our new 6.1. 

I'll close this issue in a couple of days if you've no further comments. Thanks again for reporting it!

**Comment by hussein-yassine** [comment]
Hello, I'm having the same issue on Android 8.0.0
this is the crash recorded
Unable to resume activity {com.sportsmax/com.swrve.sdk.messaging.ui.SwrveInAppMessageActivity}: java.lang.IllegalStateException: Only fullscreen activities can request orientation

**Comment by dominicmarmionswrve** [comment]
@hussein-yassine What version of the swrvesdk are you using? Are you seeing these crashes in production, and how many are you seeing? Are you able to replicate yourself? Can you show the full stacktrace please?

**Comment by hussein-yassine** [comment]
@dominicmarmionswrve  the swrve version is 8.0.0

`com.swrve.sdk.android:swrve-firebase:8.0.0`

The crashes are being seen in production yes.
I'm failing to replicate, but here's the stacktrace copied from crashlytics

Fatal Exception: java.lang.RuntimeException
Unable to resume activity {com.sportsmax/com.swrve.sdk.messaging.ui.SwrveInAppMessageActivity}: java.lang.IllegalStateException: Only fullscreen activities can request orientation
android.app.ActivityThread.performResumeActivity (ActivityThread.java:3790)
android.app.ActivityThread.handleResumeActivity (ActivityThread.java:3830)
android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:3038)
android.app.ActivityThread.handleRelaunchActivity (ActivityThread.java:4921)
android.app.ActivityThread.-wrap19 (ActivityThread.java)
android.app.ActivityThread$H.handleMessage (ActivityThread.java:1702)
android.os.Handler.dispatchMessage (Handler.java:105)
android.os.Looper.loop (Looper.java:164)
android.app.ActivityThread.main (ActivityThread.java:6944)
java.lang.reflect.Method.invoke (Method.java)
com.android.internal.os.Zygote$MethodAndArgsCaller.run (Zygote.java:327)
com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1374)

**Comment by hussein-yassine** [comment]
Hi @dominicmarmionswrve the crash still occurs on Android 8.0:
Unable to resume activity {com.sportsmax/com.swrve.sdk.messaging.ui.SwrveInAppMessageActivity}: java.lang.IllegalStateException: Only fullscreen activities can request orientation

The Project targets SDK 32, and the Swrve dependency used is: 
    implementation 'com.swrve.sdk.android:swrve-firebase:9.4.0'

**Comment by dominicmarmionswrve** [comment]
Hi @hussein-yassine, we'll take a look at this again in our next release. Thanks for letting us know.

**Comment by davidbreenwex** [comment]
Hi @hussein-yassine, 

It may be easier to solve this issue through a support ticket. 
Can you please send an email to support@swrve.com answering the following questions? 

Are you able to replicate the issue yourself? If so, what are the steps?
How many occurrences of the issue are you seeing?
Can you provide a longer stracktrace?
Are you on Android TV or Amazon?
Is there any information you can share on the devices/OS versions it's happening on?
